raycicmd/make.go

```diff
@@ -141,5 +141,11 @@ func makePipeline(repoDir string, config *config, info *buildInfo) (
 		return makeNoopBkPipeline(q), nil
 	}
 
+	if email, exists := os.LookupEnv(buildAuthorEnv); exists && config.NotifyOwnerOnFailure {
```

Comment:
don't look up envs from os here. this env should be captured in buildinfo, and set in main

we also have an interface and a fake for os envs for testing. so please don't temper with os env on testing.